src/uk/gov/hmcts/contino/YarnBuilder.groovy

```diff
@@ -314,8 +317,33 @@ EOF
   }
 
   private nagAboutOldYarnVersions() {
-     if (!isYarnV2OrNewer()){
-       WarningCollector.addPipelineWarning("old_yarn_version", "Please upgrade to Yarn V3, see https://moj.enterprise.slack.com/files/T1L0WSW9F/F04784SLAJC?origin_team=T1L0WSW9F", LocalDate.of(2023, 04, 26))
+    if (!isYarnV2OrNewer()){
+      WarningCollector.addPipelineWarning("old_yarn_version", "Please upgrade to Yarn V3, see https://moj.enterprise.slack.com/files/T1L0WSW9F/F04784SLAJC?origin_team=T1L0WSW9F", LocalDate.of(2023, 04, 26))
+    }
+  }
+
+  private isNodeJSV18OrNewer() {
+    boolean validVersion = true;
+
+    if (steps.fileExists(NVMRC)) {
+      String nodeVersion = steps.readFile(NVMRC)
+      Float current_version = valueOf(nodeVersion
+                                          .trim()
+                                          .substring(0, nodeVersion.lastIndexOf(".")))
+      steps.echo("Existing NodeJS at v${current_version}.x")
+      validVersion = current_version >= DESIRED_MIN_VERSION
+    } else {
+      steps.echo(".nvrmc file is missing for this project")
+      WarningCollector.addPipelineWarning("missing_nvrmc_file", "An nvrmc file is missing for this project. see https://github.com/hmcts/expressjs-template", NODEJS_EXPIRATION)
+    }
+
+    return validVersion
+  }
+
+  private nagAboutOldNodeJSVersions() {
+    if (!isNodeJSV18OrNewer()) {
+      steps.echo("NodeJS version is less than v18.16.0. Please update your projects .nvmrc file with the desired version")
```

Comment:
no one will see this echo, it's the WarningCollector that should tell them what to do. I would probably just remove the echo here and in the other place